src/ripple/nodestore/impl/DatabaseShardImp.cpp

```diff
@@ -1109,6 +1116,9 @@ DatabaseShardImp::initConfig(std::lock_guard<std::mutex>&)
         ledgersPerShard_ = get<std::uint32_t>(section, "ledgers_per_shard");
         if (ledgersPerShard_ == 0 || ledgersPerShard_ % 256 != 0)
             return fail("'ledgers_per_shard' must be a multiple of 256");
+
+        earliestShardIndex_ = seqToShardIndex(earliestLedgerSeq());
```

Comment:
Why are you overwriting `earliestShardIndex_` which should already be initialized in the constructor?

Reply:
Value of earliestShardIndex_ depends on parameter ledgers_per_shard which is defined to default value of 16384 in constructor. But for test we redefine ledgers_per_shard to 256. This redefinition applies to standalone mode only, so it is done not in constructor, but in init() function. After redefinition of  ledgers_per_shard, the value of earliestShardIndex_ should also be rewritten.